custom_components/plugwise/config_flow.py

```diff
@@ -285,7 +281,7 @@ async def async_step_user_gateway(self, user_input=None):
 
             if not errors:
                 await self.async_set_unique_id(
-                    api.smile_hostname or api.gateway_id, raise_on_progress=False
+                    api.smile_hostname or api.gateway_id or user_input[CONF_HOST].split(".")[0], raise_on_progress=False
```

Comment:
This added bit is not needed? The unique_id should either be the smile hostsname or the gateway_id. \We've determined in the past that these two option were sufficient to cover all known Smiles and Stretches.

Reply:
I'll investigate then, because without it and using zeroconf there was no api.smile_hostname :(

Reply:
So we should patch elsewhere then ... removing that part breaks the 'already_configured'.

Reply:
api.smile_hostname needs `connect()` to be called.

Reply:
Yeah, which is where it falls apart, that just returns `True` now, but obviously doesn't set the required parameters. I'll dig a little deeper later

Reply:
Done